metabench.cmake

```diff
@@ -152,24 +161,22 @@ file(WRITE ${METABENCH_RB_PATH}
-"          # We create exe_file and close it explicitly because if the compilation                 \n"
-"          # fails and the exe_file is deleted by the compiler, we'll raise an                     \n"
-"          # error but Tempfile.create will re-raise an error while trying to                      \n"
-"          # remove the nonexistent file.                                                          \n"
-"          exe_file = Tempfile.new(['tmp', '.o'], Dir.pwd)                                         \n"
-"          cpp_file.write(code) && cpp_file.flush                                                  \n"
-"          data = {n: n}                                                                           \n"
+"      measure_file.write(code)                                                                    \n"
+"      data = {n: n}                                                                               \n"
 "                                                                                                  \n"
-"          # Compile the file and gather timing statistics. The timing statistics                  \n"
-"          # are output to stdout when we compile the file.                                        \n"
-"          data[:time] = Benchmark.realtime {                                                      \n"
-"            command = \"#{cxx} #{cxxflags} -o #{exe_file.path} #{cpp_file.path}\"                 \n"
-"            stdout, stderr, status = Open3.capture3(command)                                      \n"
+"      # Compile the file and get timing statistics. The timing statistics                         \n"
+"      # are output to stdout when we compile the file because of the way                          \n"
+"      # the special measurement CMake target is setup.                                            \n"
+"      stdout, stderr, status = Open3.capture3(command)                                            \n"
+"      command_line = stdout.match(/\\[command line: (.+)\\]/i)                                    \n"
 "                                                                                                  \n"
-"            error_string = <<-EOS                                                                 \n"
-"compilation error: #{command}                                                                     \n"
+"      # If we didn't match anything, that's because we went too fast, CMake                       \n"
+"      # did not have the time to see the changes to the measure file and                          \n"
+"      # the target was not rebuilt. So we sleep for a bit and then retry                          \n"
+"      # this iteration.                                                                           \n"
+"      (sleep 0.2; redo) if command_line.nil?                                                      \n"
```

Comment:
The problem is that when we run the target through CMake, CMake will check whether the file is outdated too, and it will determine that it isn't. I think it has to do with the timestamps on the file not being granular enough, which is not a problem with CMake per se.